src/main/java/org/opensearch/security/action/apitokens/ApiTokenAction.java

```diff
@@ -181,39 +183,66 @@ private RestChannelConsumer handlePost(RestRequest request, NodeClient client) {
+                String name = (String) requestBody.get(NAME_FIELD);
+                long expiration = (Long) requestBody.getOrDefault(
+                    EXPIRATION_FIELD,
+                    Instant.now().toEpochMilli() + TimeUnit.DAYS.toMillis(30)
                 );
 
-                // Then trigger the update action
-                ApiTokenUpdateRequest updateRequest = new ApiTokenUpdateRequest();
-                client.execute(ApiTokenUpdateAction.INSTANCE, updateRequest, new ActionListener<ApiTokenUpdateResponse>() {
-                    @Override
-                    public void onResponse(ApiTokenUpdateResponse updateResponse) {
-                        try {
-                            XContentBuilder builder = channel.newBuilder();
-                            builder.startObject();
-                            builder.field("Api Token: ", token);
-                            builder.endObject();
-
-                            BytesRestResponse response = new BytesRestResponse(RestStatus.OK, builder);
-                            channel.sendResponse(response);
-                        } catch (IOException e) {
-                            sendErrorResponse(channel, RestStatus.INTERNAL_SERVER_ERROR, "Failed to send response after token creation");
-                        }
+                // First check token count
+                apiTokenRepository.getTokenCount(ActionListener.wrap(tokenCount -> {
+                    if (tokenCount >= 100) {
+                        sendErrorResponse(
+                            channel,
+                            RestStatus.TOO_MANY_REQUESTS,
+                            "Maximum limit of 100 API tokens reached. Please delete existing tokens before creating new ones."
+                        );
+                        return;
                     }
 
-                    @Override
-                    public void onFailure(Exception e) {
-                        sendErrorResponse(channel, RestStatus.INTERNAL_SERVER_ERROR, "Failed to propagate token creation");
-                    }
-                });
-            } catch (final Exception exception) {
-                sendErrorResponse(channel, RestStatus.INTERNAL_SERVER_ERROR, exception.getMessage());
+                    // If count is ok, create the token
+                    apiTokenRepository.createApiToken(name, clusterPermissions, indexPermissions, expiration, ActionListener.wrap(token -> {
+                        // After successful creation, trigger the update action
+                        ApiTokenUpdateRequest updateRequest = new ApiTokenUpdateRequest();
+                        client.execute(ApiTokenUpdateAction.INSTANCE, updateRequest, ActionListener.wrap(updateResponse -> {
+                            try {
+                                XContentBuilder builder = channel.newBuilder();
+                                builder.startObject();
+                                builder.field("Api Token: ", token);
```

Comment:
Let's just have the field name be `token`. No need to include the colon here because this gets serialized to json and will get included.